### DISCUSSION TITLE
Certificates created by Match have ugly metadata ("Created via API")

### DISCUSSION THREAD

**Discussion by lechuckcaptain** [fastlane/fastlane#20180] [General]
We had to recently nuke our certificates due to: https://github.com/fastlane/fastlane/issues/20164#issuecomment-1097743084

The new certificate appears to have a very ugly metadata. More in details, Name and Created By are totally undreadable.

```
Certificate Name: "Created via API"
Created By: "API Key: *********************************************************************"
```

I guess this might be related to [App Store Connect API that doesn't allow to customise this information](https://developer.apple.com/forums/thread/658791), but wanted to check with the community if I'm missing something that can be done in order to improve this situation.

**Comment by CaseyCw1** [comment]
That's great. I was impressed by your writing. I am happy to see such a topic. Please come to my blog and read it.

[GreenSkyOnline
](https://www.greenskyonline.me/)

**Comment by qupig** [comment]
Guess it might be possible to do this with `common name(CN)` in the `CSR`.

**Comment by juliengdt** [comment]
2024 and always this problem. 
Fun fact: match does it well for Distribution, (got the name of the company, whynot) but not for a Development (got always "Created via API")

Very disapointing.

Julien


EDIT: seems an Apple issue; request object never take into account such a "name". 
source: https://developer.apple.com/documentation/appstoreconnectapi/certificatecreaterequest

EDIT2: just gave an Apple technical feedback: 
FB15119608 (Unequal certificate naming convention when created with AppStore Connect API)
